### PR TITLE
[GOBBLIN-2092] updated all the columns during flow_spec update

### PR DESCRIPTION
**Pull request by pratapaditya04**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/2092)"
    - https://issues.apache.org/jira/browse/GOBBLIN-2092?j

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
Updated the SPECIFIC_INSERT_STATEMENT sql statement, to include all the columns in case of update

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java [hunk 1]
```diff
@@ -63,7 +63,8 @@ public class MysqlSpecStore extends MysqlBaseSpecStore {
   // record may contain data there... though in practice, all should, given the passage of time (amidst the usual retention expiry).
   protected static final String SPECIFIC_INSERT_STATEMENT = "INSERT INTO %s (spec_uri, flow_group, flow_name, template_uri, "
       + "user_to_proxy, source_identifier, destination_identifier, schedule, tag, isRunImmediately, owning_group, spec, spec_json) "
-      + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE spec = VALUES(spec), spec_json = VALUES(spec_json)";
+      + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE template_uri = VALUES(template_uri),user_to_proxy = VALUES(user_to_proxy),source_identifier = VALUES(source_identifier),schedule = VALUES(schedule),tag = VALUES(tag),isRunImmediately = VALUES(isRunImmediately),owning_group = VALUES(owning_group),"
```
Comment (phet) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java:66]: I wasn't familiar w/ the `ON DUPLICATE KEY` ... `VALUES()` function.  when I looked it up, it appears to be deprecated and row/column aliases are to be used instead.  see: https://dev.mysql.com/doc/refman/8.4/en/insert-on-duplicate.html

looks like `destination_identifier` may have been omitted.  keeping these in the same order as they're listed prior (right after `INSERT INTO %s`) might make that more noticeable

finally, as in java source, leave a space after `,` in SQL

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 2]
```diff
@@ -24,6 +24,8 @@
 import java.util.Iterator;
 import java.util.List;
 
+import org.apache.gobblin.runtime.api.FlowSpec;
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:27]: see import ordering here - https://gobblin.apache.org/docs/developer-guide/CodingStyle/

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 3]
```diff
@@ -24,6 +24,8 @@
...
+import org.intellij.lang.annotations.Flow;
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:28]: mistaken import?

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 4]
```diff
@@ -116,6 +135,24 @@ public void testAddSpec() throws Exception {
     Assert.assertFalse(this.specStore.exists(URI.create("dummy")));
   }
 
+  @Test
+  public void testUpdateSpec() throws Exception {
+    this.specStore.addSpec(this.flowSpec);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    Assert.assertFalse(this.specStore.exists(URI.create("dummy")));
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:143]: what are you going for here?  seems unrelated to updating specs...

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 5]
```diff
@@ -116,6 +135,24 @@ public void testAddSpec() throws Exception {
...
+
+    //Updating
+
+    this.specStore.addSpec(this.flowSpecUpdated);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    FlowSpec updated = (FlowSpec)this.specStore.getSpecImpl(this.uri1);
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("key"),"valueUpdated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("key3"),"value3UPdated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("config.with.dot"),"value4Updated");
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:153]: none of these verify any of the special "indexing" columns you specified in the `ON DUPLICATE KEY` clause, such as `user_to_proxy`, `source_identifier`, etc.

being the crux of this PR, that would seem worth adding

**Review by phet — COMMENTED** [review]
good start here!

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java (hunk 1 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java]: thanks for the detail,addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java (hunk 3 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java]: yes,removed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java (hunk 4 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java]: addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java (hunk 5 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java]: addressed

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStore.java
Comment (abhishekmjain) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStore.java:1]: Is there any change made to this file? If not, we can revert this file to original in this PR.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java
Comment (abhishekmjain) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java:1]: Can you please confirm if we would get all these values in each update call? My concern is we shouldn't be overwriting any non-updated field with blank or default value.

For example source_identifier and destination_identifier were faro initially and only destination_identifier got updated to holdem. What would happen to source_identifier? Since we are running an update query on all the fields, are we getting the entire object from DB first and then updating only modified fields?

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 6]
```diff
@@ -116,6 +136,22 @@ public void testAddSpec() throws Exception {
...
+    //Updating
+
+    this.specStore.addSpec(this.flowSpecUpdated);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    FlowSpec updated = (FlowSpec)this.specStore.getSpecImpl(this.uri1);
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.user.to.proxy"),"value1Updated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.gobblin.flow.sourceIdentifier"),"value2Updated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.gobblin.flow.destinationIdentifier"),"value3Updated");
+  }
```
Comment (abhishekmjain) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:153]: Adding a field which is unmodified and the value remains intact after update operation would be helpful.

**Comment by abhishekmjain** [comment]
Might want to squash all the commits into 1 commit. Reference: https://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java
Comment (pratapaditya04) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java:1]: No, in case of update calls, the spec in MySqlSpecStore has all the values of the Spec, so even in case if only destination_identifier got updated, the spec will contain all the original values as the patch is done on top of the original FlowSpec 
`   FlowConfig flowConfig = getFlowConfig(flowId);

    try {
      PatchApplier.applyPatch(flowConfig, flowConfigPatch);`

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 7]
```diff
@@ -89,6 +93,22 @@ public void setUp() throws Exception {
         .withVersion("Test version 2")
         .withSpecExecutor(MockedSpecExecutor.createDummySpecExecutor(new URI("execB")))
         .build();
+
+    flowSpec = new FlowSpec.Builder(this.uri1) .withConfig(ConfigBuilder.create()
+            .addPrimitive("properties.user.to.proxy", "value1")
+            .addPrimitive("properties.gobblin.flow.sourceIdentifier", "value2")
```
[the remark this answers is not in the document]
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:99]: Initializing variables in the setup method, even if they're not currently used, sets a good precedent for future updates. It promotes maintainability and makes it easier to add or modify tests later without needing to duplicate initialization code.

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java (hunk 6 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:153]: addressed

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStore.java
Comment (pratapaditya04) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStore.java:1]: addressed

**Comment by phet** [comment]
> Might want to squash all the commits into 1 commit. Reference: https://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request

the final merge into apache gobblin can squash all commits, but when working on a PR, please don't do that.  after you do, it's no longer possible for reviewers to select "all commits since [their] last review"

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 8]
```diff
@@ -116,6 +136,22 @@ public void testAddSpec() throws Exception {
...
   }
 
+  @Test
+  public void testUpdateSpec() throws Exception {
+    this.specStore.addSpec(this.flowSpec);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    //Updating
+
+    this.specStore.addSpec(this.flowSpecUpdated);
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:146]: given that `this.specStore` is a `MysqlBaseSpecStore` (on L78), I don't believe it actually uses the insert statement from `MysqlSpecStore` that you modified.

I'm open, but it would probably be cleanest if you made another test class specifically for that one.

when you do, the way to verify the "indexing columns" your new code is meaning to update would be to call `getSpecImpl(SpecSearchObject)`

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java [hunk 9]
```diff
@@ -89,6 +93,22 @@ public void setUp() throws Exception {
         .withVersion("Test version 2")
         .withSpecExecutor(MockedSpecExecutor.createDummySpecExecutor(new URI("execB")))
         .build();
+
+    flowSpec = new FlowSpec.Builder(this.uri1) .withConfig(ConfigBuilder.create()
+            .addPrimitive("properties.user.to.proxy", "value1")
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java:98]: given [how things work](https://github.com/apache/gobblin/blob/d9f83e638b6eb34fa85b0317eee31cf39c1e8546/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java#L82) in `completeInsertPreparedStatement`, I don't believe you ought to include the `properties.` prefix.  AFAIR that is only needed when passing prop updates via carbon

**Review by phet — COMMENTED** [review]
just a bit of reworking on the unit testing...

**Comment by abhishekmjain** [comment]
> > Might want to squash all the commits into 1 commit. Reference: https://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request
> 
> the final merge into apache gobblin can squash all commits, but when working on a PR, please don't do that. after you do, it's no longer possible for reviewers to select "all commits since [their] last review"

Got it, I got confused with the Commit heading in checklist as it says to squash the commits, but I didn't realize it is for the final merge.

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java (hunk 8 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java]: thanks,addressed

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java (hunk 1 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java]: the alias way is only supported on mysql 8.0.20 and above. Our test suites currently use wix-embedded-mysql and the latest mysql version it has is 8.0.17 [here](https://github.com/wix-incubator/wix-embedded-mysql/blob/master/wix-embedded-mysql/src/main/java/com/wix/mysql/distribution/Version.java)
so the test is failing due to unsupported query, as the row/column aliases way  was introduced in 8.0.20.
so have kept the ON DUPLICATE KEY ... VALUES() way of updating for now, and maybe we can create a different ticket to move away from wix-embedded-mysql and explore other alternatives. How do you suggest we go ahead?

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java [hunk 10]
```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
...
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").flowGroup("fg5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertEquals(specs.size(), 1);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().propertyFilter(FLOW_SOURCE_IDENTIFIER_KEY+"="+"sourceUpdated").build();
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java:272]: use the specific methods: e.g. the `FSSO.builder()` has a `.sourceIdentifier(String)` setter for this purpose.  (see the example using `owningGroup(String)` in the test just above.)

unit tests serve double-duty as a system spec, so it's important to show (and exercise!) the API most directly.

in addition those methods more easily enable combining multiple constraints (without delving into the CSV of `propertyFilter`).  sprinkling in an occasional combo search seems preferable to each one in isolation.

overall, do test every facet for which you fixed the "reindexing", in order to catch any error of omission.  (but no need to do one at a time)

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java [hunk 11]
```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
...
     Assert.assertTrue(specs.contains(this.flowSpec4));
   }
+  @Test
+  public void testGetSpecsAfterUpdate() throws Exception {
+    this.specStore.addSpec(this.flowSpec5);
+    FlowSpec result = (FlowSpec) this.specStore.getSpec(this.uri5);
+    removeModificationTimestampFromSpecs(result);
+    Assert.assertEquals(result, this.flowSpec5);
+    // Updating  flowSpec5
+    this.specStore.addSpec(this.flowSpec5Updated);
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java:256]: this update comes too soon, prior to thoroughly validating the "before state".

ideally both before and after would validate the same search param sets.  the majority would demonstrate "flip-flop", wherein what retrieved a result before no longer does post-update or vice-versa.  a small subset would work (or not) on both to verify intentionally unchanged/uncorrupted search facets.

one option is for the "before" validations to live in a separate method or even a heavily augmented `testGetSpec` (the one just above).  just use the same starting flowSpec there.

personally, I'd define the same `FlowSpecSearchObject` instances common to both and then use them while initializing two lists of input pairs, which I'd loop over to run through, once before update and then again after -
```
// (reappropriate `Map.Entry` as a "Pair")
List<Pair<Map.Entry<FlowSpecSearchObject, Boolean>> preUpdateValidations = ...
List<Pair<Map.Entry<FlowSpecSearchObject, Boolean>> postUpdateValidations = ...

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java [hunk 12]
```diff
@@ -61,9 +61,23 @@ public class MysqlSpecStore extends MysqlBaseSpecStore {
 
   // Historical Note: the `spec_json` column didn't always exist and was introduced for GOBBLIN-1150; the impl. thus allows that not every
   // record may contain data there... though in practice, all should, given the passage of time (amidst the usual retention expiry).
+
+  //ON DUPLICATE KEY ... VALUES() is deprecated , use alias instead https://dev.mysql.com/doc/refman/8.4/en/insert-on-duplicate.html
```
Comment (phet) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java:65]: GREAT comment! :)

**Review by phet — COMMENTED** [review]
tests are coming along

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java (hunk 11 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java]: addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java (hunk 10 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java]: addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java [hunk 13]
```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
...
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").flowGroup("fg5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertEquals(specs.size(), 1);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
```
[the remark this answers is not in the document]
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java]: addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java [hunk 14]
```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
...
+    // Updating  flowSpec5
+    this.specStore.addSpec(this.flowSpec5Updated);
+
+    FlowSpecSearchObject flowSpecSearchObject = FlowSpecSearchObject.builder().flowGroup("fg5").build();
+    Collection<Spec>  specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
+
+    flowSpecSearchObject = FlowSpecSearchObject.builder().flowName("fn5").build();
+    specs = this.specStore.getSpecs(flowSpecSearchObject);
+    Assert.assertTrue(specs.contains(this.flowSpec5Updated));
```
[the remark this answers is not in the document]
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java]: addressed

File: gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java [hunk 15]
```diff
@@ -130,6 +132,35 @@ public void setUp() throws Exception {
...
+
+    flowSpec5Updated=
+        FlowSpec.builder(this.uri5)
+            .withConfig(ConfigBuilder.create()
+                .addPrimitive(FLOW_SOURCE_IDENTIFIER_KEY, "sourceUpdated")
+                .addPrimitive(FLOW_DESTINATION_IDENTIFIER_KEY, "destinationUpdated")
+                .addPrimitive(ConfigurationKeys.JOB_SCHEDULE_KEY, "0 0 9 * * ? *")
+                .addPrimitive(ConfigurationKeys.FLOW_OWNING_GROUP_KEY, "owningGroupUpdated")
+                .addPrimitive(ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "true")
+                .addPrimitive("user.to.proxy", "userAUpdated")
```
[the remark this answers is not in the document]
Comment (pratapaditya04) [reply gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java]: Sure, have kept FLOW_RUN_IMMEDIATELY unupdated,

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java (hunk 1 above)
Comment (pratapaditya04) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java]: Have updated the mysql test version in this [PR](https://github.com/apache/gobblin/pull/3986)
